presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveStorageFormats.java

```diff
@@ -185,6 +200,37 @@ public void testInsertIntoPartitionedTable(StorageFormat storageFormat)
         query(format("DROP TABLE %s", tableName));
     }
 
+    @Test(dataProvider = "storage_formats_with_null_format", groups = STORAGE_FORMATS)
+    public void testInsertIntoTableWithNullFormat(StorageFormat storageFormat)
+    {
+        String tableName = format("storage_format_%s_with_null_format", storageFormat.getName());
+        String nullFormat = "null_value";
+
+        query(format("CREATE TABLE %s (name VARCHAR) " +
+                        "WITH (format = '%s', null_format = '%s')",
+                tableName,
+                storageFormat.getName(),
+                nullFormat));
+
+        // \N is the default null format
+        String[] values = new String[] {nullFormat, null, "non-null", "", "\\N"};
+
+        Row[] storedValues = Arrays.stream(values).map(Row::row).toArray(Row[]::new);
+        // If you try to put the null format in a table, it will come out as null.
+        storedValues[0] = row((Object) null);
+
+        String placeholders = String.join(", ", Collections.nCopies(values.length, "(?)"));
+
+        query(format("INSERT INTO %s VALUES %s", tableName, placeholders),
+                Arrays.stream(values)
+                        .map(value -> param(JDBCType.VARCHAR, value))
+                        .toArray(QueryParam[]::new));
+
+        assertThat(query(format("SELECT * FROM %s", tableName))).containsOnly(storedValues);
+
+        onHive().executeQuery(format("DROP TABLE %s", tableName));
```

Comment:
put this one in `finally`

Reply:
Almost all of the product tests don't include that step; should I add it throughout this file?

Reply:
No never mind. Keep it as is. It does not matter much and code in finally can actually make it harder to debug test by masking the exception which caused original failure.